PhpcrMigration/Application/Parser/ArticleNodeParser.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PhpcrMigrationBundle\PhpcrMigration\Application\Parser;
+
+use PHPCR\NodeInterface;
+use Sulu\Bundle\PhpcrMigrationBundle\PhpcrMigration\Application\Persister\AbstractPersister;
+use Sulu\Bundle\PhpcrMigrationBundle\PhpcrMigration\Application\Repository\EntityRepositoryInterface;
+
+class ArticleNodeParser implements NodeParserInterface
+{
+    public function __construct(private readonly EntityRepositoryInterface $repository)
+    {
+    }
+
+    public function parse(NodeInterface $node): array
+    {
+        if (!$this->supports($node)) {
+            return [];
+        }
+
+        $localizations = [];
+        $localizations = $this->parseLocalizedRoutes($node, $localizations);
+
+        return [
+            'localizations' => $localizations,
+        ];
+    }
+
+    private function supports(NodeInterface $node): bool
+    {
+        foreach ($node->getMixinNodeTypes() as $mixinNodeType) {
+            if ('sulu:article' == $mixinNodeType->getName()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array<string, array<string, mixed>> $localizations
+     *
+     * @return array<string, array<string, mixed>>
+     */
+    private function parseLocalizedRoutes(NodeInterface $node, array $localizations): array
+    {
+        $routes = $this->repository->findBy(
+            AbstractPersister::LEGACY_ROUTE_TABLE,
+            [
+                'entity_class' => 'Sulu\Bundle\ArticleBundle\Document\ArticleDocument',
```

Comment:
@alexander-schranz Not sure if this is always the ArticleDocument or if the entity_class is changed when you extend the ArticleDocument

Reply:
I think as the uuid is unique and the class can be extended we can go just on the uuid itself.